src/main/java/org/folio/circulation/domain/OverdueFineCalculatorService.java

```diff
@@ -52,39 +60,43 @@ private OverdueFineCalculatorService(Clients clients) {
       new ItemRepository(clients, true, false, false),
       new FeeFineOwnerRepository(clients),
       new FeeFineRepository(clients),
-      new OverduePeriodCalculatorService(new CalendarRepository(clients)));
+      new OverduePeriodCalculatorService(new CalendarRepository(clients)),
+      new UserRepository(clients),
+      new FeeFineActionRepository(clients));
   }
 
   public CompletableFuture<Result<CheckInProcessRecords>> calculateOverdueFine(
-    CheckInProcessRecords records) {
+    CheckInProcessRecords records, WebContext context) {
 
     Loan loan = records.getLoan();
 
-    if (loan == null) {
+    if (loan == null || context == null) {
       return completedFuture(of(() -> records));
     }
 
-    return calculateOverdueFine(loan).thenApply(mapResult(v -> records));
+    return calculateOverdueFine(loan, context.getUserId()).thenApply(mapResult(v -> records));
```

Comment:
Consider using `records.withLoggedInUserId()` but make sure that it is populated in the calling method.

Reply:
There are two places this calculator can be called from - check-in and renewal (hence the two `calculateOverdueFine()` methods). From check-in it receives `CheckInProcessRecords` which does have `getLoggedInUserId()` method - I tried to use it, this field is in fact populated. But from renewal it receives `LoanAndRelatedRecords` which doesn't have logged in user info, so I decided to pass it directly. I can try to extend `LoanAndRelatedRecords` if that's a preferable approach, but from the name it looks like it's not necessarily supposed to exist within a single process initiated by a single user.